predicators/approaches/bridge_policy_approach.py

```diff
@@ -486,13 +551,28 @@ def learn_from_interaction_results(
             return None
         all_states = []
         all_actions = []
-
-        for result in results:
-            new_traj = LowLevelTrajectory(result.states, result.actions)
+        policy_logs = self._policy_logs
+        for i in range(len(results)):
+            result = results[i]
+            policy_log = policy_logs[:len(result.states[:-1])]
+            mapleq_states = [
+                state for j, state in enumerate(result.states[:-1])
+                if policy_log[j] == "bridge"
+                or policy_log[max(j - 1, 0)] == "bridge"
+            ]
+            mapleq_actions = [
+                action for j, action in enumerate(result.actions)
+                if policy_log[j] == "bridge"
+                or policy_log[max(j - 1, 0)] == "bridge"
+            ]
+            mapleq_states.append(result.states[-1])
+            new_traj = LowLevelTrajectory(mapleq_states, mapleq_actions)
             self._trajs.append(new_traj)
-            all_states.extend(result.states)
-            all_actions.extend(result.actions)
+            all_states.extend(mapleq_states)
+            all_actions.extend(mapleq_actions)
+            policy_logs = policy_logs[len(result.states) - 1:]
 
         self.mapleq.get_interaction_requests()
         self.mapleq._learn_nsrts(self._trajs, 0, [] * len(self._trajs))  # pylint: disable=protected-access
+        self._policy_logs = []
         return None
```

Comment:
could you remove the `return None` here?

Reply:
the type checker gets mad without the return None :,)

Reply:
oh gotcha, nw